<launch>
    
    <param name="/use_sim_time" value="ture"/>
    <node pkg="rosbag" type="play" name="playbag" args="--clock /home/ziv/sunshine_dateset/urban07.bag
         /rslidar_points:=/left/velodyne_points
         /ns1/rslidar_points:=/left/velodyne_points
         /ns2/rslidar_points:=/right/velodyne_points
         /ns1/velodyne_points:=/left/velodyne_points
         /ns2/velodyne_points:=/right/velodyne_points
    "/>

    <param name="file_save_path" type="string" value="/home/ziv/mloam/" />

    <param name="scan_line" type="int" value="16" />
    <param name="min_dis" type="double" value="0.1" />
    <param name="max_dis" type="double" value="100.0" />
    <param name="map_resolution" type="double" value="0.4" />
    <param name="submap_len" type="int" value="10" />
    <param name="keyframe_dist_threshold" type="double" value="3" />
    <param name="keyframe_angle_threshold" type="double" value="0.1" />
    <param name="save_map_resolution" type="double" value="0.2" />
    <param name="gicp_fitness_socre" type="double" value="0.5" />
    <param name="need_loop" type="bool" value="true" />
    <param name="loop_keyframe_crop_len" type="int" value="6" />
    <param name="loop_close_distance" type="int" value="10" />
    <param name="loop_cooldown" type="int" value="15" />

    <!--    calibration part     -->
    <param name="t_0_1_z_prior" type="double" value="-0.095" />

    <node pkg="mloam" type="laserProcessing" name="laserProcessing" output="screen" />

    <node pkg="mloam" type="laserOdometry" name="laserOdometry" output="screen" />

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find mloam)/rviz/show.rviz" />
    </group>

</launch>
